probe-rs/src/probe/cmsisdap/mod.rs

```diff
@@ -809,6 +824,8 @@ impl DebugProbe for CmsisDap {
             // to ensure the debug port is ready for JTAG signals,
             // at which point we can interrogate the scan chain
             // and configure the probe with the given IR lengths.
+            self.scan_chain()?;
+            self.select_target(0)?;
```

Comment:
Considering the comment above, is this really the best place to do this? I know the other drivers do similar things, but I'm not sure 🤔

If the code stays, the comment is incorrect. On the other hand, this must not break SWJ, and I don't know anything about SWJ to know whether this addition is a problem.

Reply:
This operation should probably be implemented in the architecture, but I'm not familiar with it, so I refer to other probe jtag implementations